packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -219,282 +206,76 @@ export class OdspDocumentService implements IDocumentService {
-            if (finalWebsocketToken === null) {
-                throw this.annotateConnectionError(
-                    new NonRetryableError(
-                        "Websocket token is null",
-                        OdspErrorType.fetchTokenError,
-                        { driverVersion },
-                    ),
-                    "getWebsocketToken",
-                    !requestWebsocketTokenFromJoinSession);
-            }
-            try {
-                const connection = await this.createDeltaConnection(
-                    websocketEndpoint.tenantId,
-                    websocketEndpoint.id,
-                    finalWebsocketToken,
-                    io,
-                    client,
-                    websocketEndpoint.deltaStreamSocketUrl);
-                connection.on("op", (documentId, ops: ISequencedDocumentMessage[]) => {
-                    this.opsReceived(ops);
-                });
-                // On disconnect with 401/403 error code, we can just clear the joinSession cache as we will again
-                // get the auth error on reconnecting and face latency.
-                connection.once("disconnect", (error: any) => {
-                    // Clear the join session refresh timer so that it can be restarted on reconnection.
-                    this.clearJoinSessionTimer();
-                    if (typeof error === "object" && error !== null
-                        && error.errorType === DriverErrorType.authorizationError) {
-                        this.cache.sessionJoinCache.remove(this.joinSessionKey);
-                    }
-                    // If we hit this assert, it means that "disconnect" event is emitted before the connection went through
-                    // dispose flow which is not correct and could lead to a bunch of erros.
-                    assert(connection.disposed, 0x4ae /* Connection should be disposed by now */);
-                    this.currentConnection = undefined;
-                });
-                this.currentConnection = connection;
-                return connection;
-            } catch (error) {
-                this.cache.sessionJoinCache.remove(this.joinSessionKey);
-
-                const normalizedError = this.annotateConnectionError(
-                    error,
-                    "createDeltaConnection",
-                    !requestWebsocketTokenFromJoinSession);
-                if (typeof error === "object" && error !== null) {
-                    normalizedError.addTelemetryProperties({ socketDocumentId: websocketEndpoint.id });
-                }
-                throw normalizedError;
-            }
-        });
-    }
-
-    private clearJoinSessionTimer() {
-        if (this.joinSessionRefreshTimer !== undefined) {
-            clearTimeout(this.joinSessionRefreshTimer);
-            this.joinSessionRefreshTimer = undefined;
+        if (this.socketModuleP === undefined) {
+            this.socketModuleP = this.getDelayLoadedDeltaStream();
```

Comment:
>this.getDelayLoadedDeltaStream(); [](http://example.com/codeflow?start=33&length=33)

Can it return undefined ? Should we assert if odspDelayLoadedDeltaStream.ts somehow is not found or fails to be imported ?

Reply:
If you look in this function, I throw on socket module import failure.